internal/broker/broker.go

```diff
@@ -778,7 +778,10 @@ func (b *Broker) CancelIsAuthenticated(sessionID string) {
 func (b *Broker) UserPreCheck(username string) (string, error) {
 	found := false
 	for _, suffix := range b.cfg.allowedSSHSuffixes {
-		if strings.HasSuffix(username, suffix) {
+		if suffix == "" {
+			continue
+		}
+		if suffix == "*" || strings.HasSuffix(username, suffix) {
```

Comment:
One thing that was brought up when we discussed this in yesterday's standup is that if we support the special value `*`, users will assume that they can also use `*@foo.org` to match all usernames ending with `@foo.org`.

We could use  https://github.com/gobwas/glob to also support that syntax.

Reply:
mmh, if we support globs, it would make more sense to name the option `allowed_ssh_users` than `allowed_ssh_suffixes`, because you can configure patterns which do not only match suffixes

Reply:
Definitely in favor of keeping it simple. Full glob support (and renaming the setting accordingly) now also seems too much to me. However, supporting a leading `*`, i.e. `*@domain.com` in addition to `@domain.com`, seems simple enough to me that I would prefer to support it (via the small diff I pasted).

Reply:
I don't like the idea of having both `*`, `*@domain.com`, and `@domain.com` as valid values. I think it might be more harmful than helpful for the understanding here. What if we stop naming it as `suffixes` and use something like `allowed_domains` instead?

Reply:
What do you see as potentially harmful in supporting both `*@domain.com` and `@domain.com`? I genuinely want to understand.

In case that there's a misunderstanding, I only want to support the case that the prefix *starts* with a single `*`. I don't want to support the case that `*` appears anywhere else but the start. I wouldn't even document `*@domain.com`, it's just meant to gracefully handle the case that users (understandably, IMO) use `*@domain.com` if they see that `*` is supported.